sqlglot/schema.py

```diff
@@ -149,6 +217,35 @@ def _nested_get(d, *path):
     return d
 
 
+def _nested_set(d, keys, value):
+    """
+    In-place set a value for a nested dictionary
+
+    Ex:
+        >>> _nested_set({}, ["top_key", "second_key"], "value")
+        {'top_key': {'second_key': 'value'}}
+        >>> _nested_set({"top_key": {"third_key": "third_value"}}, ["top_key", "second_key"], "value")
+        {'top_key': {'third_key': 'third_value', 'second_key': 'value'}}
+
+    d (dict): dictionary
+    keys (Iterable[str]): ordered iterable of keys that makeup path to value
+    value (Any): The value to set in the dictionary for the given key path
+    """
+    if not keys:
+        return
+    if len(keys) == 1:
+        d[keys[0]] = value
+        return
+    subd = d
+    for key in keys[:-1]:
+        if key not in subd:
+            subd = subd.setdefault(key, {})
+        else:
+            subd = subd[key]
```

Comment:
what if the target key is not a dict?

Reply:
We know that it will be given the structure of the schema mapping we are using in this file. A good reason why this function exists here and not as a general utility function because of assumptions like that.